src/client/bedrock_client.rs

```diff
@@ -29,52 +29,102 @@ impl BedrockClient {
+                    "dimensions": dimension,
+                })
+            } else {
+                json!({
+                    "inputText": text.to_string(),
+                })
             };
 
             let result = match client
                 .invoke_model()
-                .model_id("amazon.titan-embed-text-v2:0")
+                .model_id(model.clone())
                 .content_type("application/json")
                 .body(Blob::new(serde_json::to_string(&prompt).unwrap()))
                 .send()
                 .await
             {
                 Ok(r) => r,
                 Err(e) => {
-                    return Err(ShellError::GenericError {
-                        error: format!("error returned from Bedrock API: {:?}", e),
-                        msg: "".to_string(),
-                        span: None,
-                        help: Some("Please supply AWS SDK config and credentials in ~/.aws/config and ~/.aws/credentials files".to_string()),
-                        inner: Vec::new(),
-                    })
+                    match e {
+                        aws_smithy_runtime_api::client::result::SdkError::DispatchFailure(_) => {
+                            return Err(ShellError::GenericError {
+                                error: "failed to dispatch Bedrock embedding request".to_string(),
+                                msg: "".to_string(),
+                                span: None,
+                                help: Some(
+                                    "check aws credentials are correctly configured".to_string(),
+                                ),
+                                inner: Vec::new(),
+                            });
+                        }
+                        aws_smithy_runtime_api::client::result::SdkError::ServiceError(err) => {
+                            let (err_msg, help) = match err.err() {
+                                InvokeModelError::ResourceNotFoundException(inner_err) => {
+                                    (inner_err.message.as_ref().unwrap().to_string(),
+                                    Some("Supply the name of the model as you would in a Bedrock API request.".to_string()))
+                                },
+                                InvokeModelError::AccessDeniedException(inner_err) => {
+                                    (inner_err.message.as_ref().unwrap().to_string(),
+                                    Some("Have you been granted access to this model in the AWS web console?".to_string()))
+                                },
+                                InvokeModelError::ValidationException(inner_err) => {
+                                    let msg = inner_err.message.as_ref().unwrap().to_string();
+                                    let help = None;
+
+                                    let (msg, help) = if msg.contains("extraneous key [dimensions] is not permitted") {
```

Comment:
I guess there's no error codes we can use? Just aware this is going to break as soon as they change the text.

Reply:
None that I could find unfortunately, I know it fragile but I think this is the best we can do.

Reply:
Are the messages the provider give us actually fairly descriptive anyway? Thinking that users might have more lucky googling those error messages if they need to.

Reply:
Yeah you're right, this is the result of not doing the string matching: 
```
> vector enrich-text testing --model another-llm-model
Embedding batch 1/1
Error:   × The provided model identifier is invalid.
  help: Check that the model name is correct and that it supports the chosen dimensionality.

> vector enrich-text testing --dimension 1000000 --model amazon.titan-embed-text-v2
Embedding batch 1/1
Error:   × The provided model identifier is invalid.
  help: Check that the model name is correct and that it supports the chosen dimensionality.
  
  > vector enrich-text testing --dimension 1000000 --model amazon.titan-embed-text-v2:0
Embedding batch 1/1
Error:   × Malformed input request: #: only 1 subschema matches out of 2, please reformat your input and try again.
  help: Check that the model name is correct and that it supports the chosen dimensionality.
```